workflow_test.go

```diff
@@ -0,0 +1,274 @@
+
+	t.Run("TestTrigger", func(t *testing.T) {
+		p := workflow.Params{
+			Owner:       *owner,
+			Repo:        *repo,
+			AccessToken: *accessToken,
+			EventType:   *eventType,
+			BaseURL:     baseURL,
+		}
+		w := workflow.Configure(p)
+
+		got := w.Trigger()
+
+		if got != nil {
+			t.Errorf("expecting nil err, got %v", got)
+		}
+	})
+
+	t.Run("TestTriggerContext", func(t *testing.T) {
+		p := workflow.Params{
+			Owner:       *owner,
+			Repo:        *repo,
+			AccessToken: *accessToken,
+			EventType:   *eventType,
+			BaseURL:     baseURL,
+		}
+
+		w := workflow.Configure(p)
+
+		ctx := context.Background()
+		got := w.TriggerContext(ctx)
+
+		if got != nil {
+			t.Errorf("expecting nil err, got %v", got)
+		}
+	})
+}
+
+func TestWorkflowNotFound(t *testing.T) {
+	server := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusNotFound)
+	}))
+	defer server.Close()
+
+	baseURL := strings.Replace(server.URL, "https://", "", 1)
+
+	t.Run("TestTriggeNotFound", func(t *testing.T) {
+		p := workflow.Params{
+			Owner:       *owner,
+			Repo:        "unknown-repo",
+			AccessToken: *accessToken,
+			EventType:   *eventType,
+			BaseURL:     baseURL,
+		}
+		w := workflow.Configure(p)
+
+		got := w.Trigger()
+
+		if got == nil {
+			t.Fail()
+		}
+	})
+
+	t.Run("TestTriggerContextNotFound", func(t *testing.T) {
+		p := workflow.Params{
+			Owner:       *owner,
+			Repo:        "unknown-repo",
+			AccessToken: *accessToken,
+			EventType:   *eventType,
+			BaseURL:     baseURL,
+		}
+		w := workflow.Configure(p)
+
+		ctx := context.Background()
+		got := w.TriggerContext(ctx)
+
+		if got == nil {
+			t.Fail()
+		}
+	})
+}
+
+func TestWorkflowUnauthorized(t *testing.T) {
+	server := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusUnauthorized)
+
+		errResponse := map[string]string{
+			"message": "Bad credentials",
+		}
+
+		json.NewEncoder(w).Encode(errResponse)
+	}))
+	defer server.Close()
+
+	baseURL := strings.Replace(server.URL, "https://", "", 1)
+	expected := "Bad credentials"
+
+	t.Run("TestTriggerUnauthorized", func(t *testing.T) {
+		p := workflow.Params{
+			Owner:       *owner,
+			Repo:        "unknown-repo",
+			AccessToken: *accessToken,
+			EventType:   *eventType,
+			BaseURL:     baseURL,
+		}
+		w := workflow.Configure(p)
+
+		got := w.Trigger()
+
+		if got == nil {
+			t.Fail()
+		}
+
+		if got.Error() != expected {
+			t.Errorf("expected an error message: %s, got: %s", expected, got.Error())
+		}
+	})
+
+	t.Run("TestTriggerContextUnauthorized", func(t *testing.T) {
+		p := workflow.Params{
+			Owner:       *owner,
+			Repo:        "unknown-repo",
+			AccessToken: *accessToken,
+			EventType:   *eventType,
+			BaseURL:     baseURL,
+		}
+		w := workflow.Configure(p)
+
+		ctx := context.Background()
+		got := w.TriggerContext(ctx)
+
+		if got == nil {
+			t.Fail()
```

Comment:
I think you may want to add a message here, like "Did not error as expected", otherwise how is the test runner going to know what went wrong?